akka-http-core/src/main/scala/akka/http/scaladsl/settings/ConnectionPoolSettings.scala

```diff
@@ -21,6 +24,26 @@ object PoolImplementation {
   case object New extends PoolImplementation
 }
 
+@ApiMayChange
+trait HostOverride {
+  def apply(host: String, settings: ConnectionPoolSettings): ConnectionPoolSettings
+}
+
+@InternalApi
+private[akka] object NoOpHostOverride extends HostOverride {
+  override def apply(host: String, settings: ConnectionPoolSettings): ConnectionPoolSettings = settings
+}
+
+@InternalApi
+private[akka] object DefaultHostOverride extends HostOverride {
+  override def apply(host: String, settings: ConnectionPoolSettings): ConnectionPoolSettings = {
+    if (settings.hostOverrides.isEmpty) settings
+    else {
+      settings.hostOverrides.find(_._1.pattern.matcher(host).matches()).map(_._2).getOrElse(settings)
```

Comment:
`find` means we take the first override which only makes sense if the ordering is specified.

So, I think we still need to resolve the issue about what should happen if multiple clauses match a target host (the original author tried a solution for "cascading overrides" but found it too complex in the end). As a good-enough first solution we could either ignore overrides if multiple overrides are matching, or make sure that clauses are really in a specified order and then define that the first clause will be taken.

Reply:
hostOverrides are a list in the configuration and parsed to a `Seq`, so perhaps it would be sufficient to document that we select the first matching override?

Reply:
Ah, but unfortunately typesafe-config doesn't keep the order of maps (mainly because of what the semantics should be when merging those, https://github.com/lightbend/config/issues/365). Perhaps we should consider changing to a list after all...